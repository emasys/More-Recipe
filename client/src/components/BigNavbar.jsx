import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import ReactTooltip from 'react-tooltip';
import PropTypes from 'prop-types';

// components
import Auth from './auth';

const logout = () => {
  Auth.logout();
};

console.log("is logged in", Auth.loggedIn());

/**
 *
 *
 * @class Navbar
 * @extends {Component}
 */
const Navbar = () => (
  <section className="container-fluid m-0 p-0">
    <nav
      data-aos="flip-up"
      className="navbar bg-transparent navbar-expand navbar-dark"
      style={{ zIndex: 1000, height: '200px' }}
    >
      <div className="container">
        <Link className="navbar-brand bolder ml-3 text-orange" to="/">
          <span className="nb">More Recipes</span>
        </Link>
        <button
          className="navbar-toggler"
          type="button"
          data-toggle="collapse"
          data-target="#navbarSupportedContent"
          aria-controls="navbarSupportedContent"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <i className="fa fa-bars fa-2x text-orange" aria-hidden="true" />
        </button>
        <div
          className="collapse navbar-collapse justify-content-end"
          id="navbarSupportedContent"
        >
          <ul className="navbar-nav bold">
            {Auth.loggedIn() ? (
              <li className="nav-item mt-5 pr-15 ml-3">
                <Link className=" text-orange" to={`/profile/${Auth.userID()}`}>
                  Hi {Auth.moniker()},
                </Link>
              </li>
            ) : (
              <li className="nav-item mt-5 pr-15 ml-3">
                <Link className=" text-orange" to="/">
                  Hi there,
                </Link>
              </li>
            )}
            <li className="nav-item">
              {Auth.loggedIn() ? (
                <a
                  onClick={logout}
                  className="nav-link btn btn-outline-light btn-lg btn-sign "
                  href="/"
                >
                  sign out
                </a>
              ) : (
                <Link
                  className="nav-link btn btn-outline-light btn-lg btn-sign"
                  to="/signin"
                >
                  sign in
                </Link>
              )}
            </li>
          </ul>
          <ReactTooltip place="bottom" type="dark" effect="float" />
        </div>
      </div>
    </nav>
  </section>
);

export default Navbar;
