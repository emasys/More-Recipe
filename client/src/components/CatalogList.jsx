import { Link } from 'react-router-dom';
import approx from 'approximate-number';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import React from 'react';

/**
 *
 *
 * @param {number} id
 * @returns {object} toggle classes on mouse in
 */
const onHoverIn = id => {
  document.querySelector(`#recipe-${id}`).classList.remove('crop-text');
  document.querySelector(`#recipe-${id}`).classList.add('full-text');
  document.querySelector(`#img-${id}`).classList.add('d-half');
  document.querySelector(`#tag-${id}`).classList.add('d-none');
};

/**
 *
 *
 * @param {number} id
 * @returns {object} toggle classes on mouse out
 */
const onHoverOut = id => {
  document.querySelector(`#img-${id}`).classList.remove('d-half');
  document.querySelector(`#tag-${id}`).classList.remove('d-none');
  document.querySelector(`#recipe-${id}`).classList.remove('full-text');
  document.querySelector(`#recipe-${id}`).classList.add('crop-text');
};
/**
 *
 *
 * @param {object} props
 * @returns {object} list of recipes
 */
const generateList = props => {
  if (props.catalog) {
    if (props.catalog.length < 1) {
      if (props.isLoading) {
        return (
          <div className="text-center error-message">
            <div>
              <img
                className="img-fluid"
                src="https://res.cloudinary.com/emasys/image/upload/v1516439649/mR_2_jwnuce.png"
                alt="logo"
                height="200"
                width="200"
              />
              <h4 className="p-3 m-2 text-center">...Fetching Recipes</h4>
              <img
                src="https://res.cloudinary.com/emasys/image/upload/v1516647862/Facebook-0.9s-200px_sqqnu9.gif"
                width="100"
                height="100"
                alt="loading..."
              />
            </div>
          </div>
        );
      }
      return (
        <div className="text-center error-message">
          <div>
            <img
              className="img-fluid"
              src="https://res.cloudinary.com/emasys/image/upload/v1516439649/mR_2_jwnuce.png"
              alt="logo"
              height="200"
              width="200"
            />
            <h4 className="p-3 m-2 text-center">...Oops</h4>
            <p className="p-3 m-2 text-center">No recipe here</p>
            <p className="p-3 m-2 text-center">
              <Link to="/" className="btn btn-outline-dark hvr-icon-back">
                ...go back home
              </Link>
            </p>
          </div>
        </div>
      );
    }
    return props.catalog.map(item => (
      <div className="row" key={item.id}>
        <div className="col-lg-12 col-sm-12 mb-20 mt-50 col-md-12">
          <div>
            <Link to={`/recipe/${item.id}`} className="hvr-grow-shadow">
              <div
                className="card"
                data-aos="fade-up"
                data-aos-duration="1000"
                key={item.id}
                onMouseEnter={() => onHoverIn(item.id)}
                onMouseLeave={() => onHoverOut(item.id)}
              >
                {props.auth.authInfo.userId === item.userId &&
                  props.showDeleteBtn && (
                    <button
                      className="btn btn-danger btn-sm delete-btn"
                      onClick={event => props.deleteRecipe(event, item)}
                      data-toggle="modal"
                      data-target="#deleteModal"
                    >
                      Delete Recipe
                    </button>
                  )}
                <div id={`img-${item.id}`}>
                  <img
                    className="card-img-top img-box "
                    src={item.foodImg}
                    alt="recipeImage"
                  />
                </div>

                <div className="card-body p-0 text-center social-icons">
                  <span className="tag bg-danger" id={`tag-${item.id}`}>
                    {item.category}
                  </span>
                  <h4 className="card-title custom-bg bg-dark p-2 m-0">
                    {item.name.length > 25 ?
                      item.name.slice(0, 24).concat('...') :
                      item.name}
                  </h4>
                  <div className="card-body p-5 text-left bg-wheat">
                    <p id={`recipe-${item.id}`} className="crop-text">
                      {item.description}
                    </p>
                  </div>
                  <span>
                    <i className="fa fa-heart-o" aria-hidden="true" />
                    {approx(item.favorite)}
                  </span>
                  <span>
                    <i className="fa fa-thumbs-o-up" aria-hidden="true" />
                    {approx(item.upvote)}
                  </span>
                  <span>
                    <i className="fa fa-thumbs-o-down" aria-hidden="true" />
                    {approx(item.downvote)}
                  </span>
                  <span>
                    <i className="fa fa-eye" aria-hidden="true" />
                    {approx(item.views)}
                  </span>
                  <span>
                    <i className="fa fa-comment-o" aria-hidden="true" />
                    {approx(item.comments)}
                  </span>
                </div>
              </div>
            </Link>
          </div>
        </div>
      </div>
    ));
  }
};

/**
 *
 *
 * @param {object} props
 * @returns {object} container function
 */
const CatalogList = props => (
  <div className="row justify-content-center">{generateList(props)}</div>
);

generateList.propTypes = {
  showDeleteBtn: PropTypes.bool.isRequired,
  catalog: PropTypes.object.isRequired,
  auth: PropTypes.object.isRequired,
  isLoading: PropTypes.bool.isRequired
};

const mapStateToProps = state => ({
  isLoading: state.isLoading,
  auth: state.user
});
export default connect(mapStateToProps, null)(CatalogList);
