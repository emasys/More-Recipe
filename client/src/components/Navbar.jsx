import React, { Component } from 'react';
import { Link, NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import ReactTooltip from 'react-tooltip';
import PropTypes from 'prop-types';

// actions
import { getProfile } from '../actions/userActions';
import { signOut } from '../actions/authActions/index';

// components
import Navlinks from './Navlinks';
import config from '../config';

/**
 *
 *
 * @class Navbar
 * @extends {Component}
 */
export class Navbar extends Component {
  static propTypes = {
    user: PropTypes.object,
    getProfile: PropTypes.func.isRequired,
    className: PropTypes.string.isRequired,
    auth: PropTypes.object.isRequired
  };

  static defaultProps = {
    user: null
  }
  /**
   * Creates an instance of Navbar.
   * @param {object} props
   * @memberof Navbar
   */
  constructor(props) {
    super(props);

    this.state = {
      avatar: null
    };
  }
  /**
   *
   *
   * @memberof Navbar
   *
   * @returns {object}
   * invoked immediately after a component is mounted
   */
  componentDidMount = () => {
    if (this.props.auth.isLoggedIn && !this.props.user) {
      this.props.getProfile(this.props.auth.authInfo.userId);
    } else if (this.props.auth.isLoggedIn && this.props.user) {
      this.setState({
        avatar: this.props.user.data.avatar
      });
    }
  };
  /**
   *
   * invoked before a mounted component receives new props.
   * @param {object} nextProps
   * @memberof Navbar
   * @returns {void}
   */
  componentWillReceiveProps = nextProps => {
    if (nextProps.user) {
      this.setState({
        avatar: nextProps.user.data.avatar
      });
    }
  };
  /**
   * Sign a user out of the app
   *
   * @returns {void}
   *
   * @param {object} event
   *
   * @memberOf Navbar
   */
  logOut = () => {
    this.props.signOut();
  }
  /**
   *
   * @returns {JSX.Element}
   * render react element into the DOM
   * @memberof Navbar
   */
  render() {
    return (
      <section className="container-fluid m-0 p-0">
        <nav
          data-aos="flip-up"
          className={`navbar navbar-expand-lg navbar-dark ${
            this.props.className
          }`}
        >
          <div className="container">
            <Link className="navbar-brand bolder ml-3 text-orange" to="/">
              <span className="nb">More Recipes</span>
            </Link>
            <button
              className="navbar-toggler"
              type="button"
              data-toggle="collapse"
              data-target="#navbarSupportedContent"
              aria-controls="navbarSupportedContent"
              aria-expanded="false"
              aria-label="Toggle navigation"
            >
              <i className="fa fa-bars fa-2x text-orange" aria-hidden="true" />
            </button>
            <div
              className="collapse navbar-collapse justify-content-end"
              id="navbarSupportedContent"
            >
              <ul className="navbar-nav">
                <li>
                  <NavLink
                    className="nav-link "
                    activeClassName="active"
                    to="/catalog"
                    data-tip="Search for recipes"
                  >
                    <i className="material-icons fa-2x d-none d-lg-inline">
                      &#xE8B6;
                    </i>
                    <span className="d-lg-none pl-3">Search</span>
                  </NavLink>
                </li>
                <li className="nav-item">
                  <NavLink
                    className="nav-link"
                    activeClassName="active"
                    to="/catalog"
                    data-tip="Catalog"
                  >
                    <i
                      className="material-icons fa-2x  d-none d-lg-inline"
                      aria-hidden="true"
                    >
                      &#xE8EF;
                    </i>
                    <span className="d-lg-none pl-3">Catalog</span>
                  </NavLink>
                </li>
                <li className="nav-item">
                  <NavLink
                    className="nav-link "
                    activeClassName="active"
                    to="/favorites"
                    data-tip="Your favorites"
                    id="favorite"
                  >
                    <i className="material-icons fa-2x red d-none d-lg-inline">
                      &#xE87D;
                    </i>
                    <span className="d-lg-none pl-3">Favorites</span>
                  </NavLink>
                </li>
                <li className="nav-item dropdown">
                  {this.props.auth.isLoggedIn ? (
                    <a
                      className="nav-link dropdown-toggle"
                      href="#"
                      id="navbarDropdownMenuLink"
                      data-toggle="dropdown"
                      aria-haspopup="true"
                      aria-expanded="false"
                    >
                      <img
                        src={
                          this.state.avatar || config.DEFAULT_DISPLAY_PICTURE
                        }
                        alt="avi"
                        className="fa-2x img-icon rounded-circle"
                      />
                    </a>
                  ) : (
                    <a
                      className="nav-link "
                      href="#"
                      id="navbarDropdownMenuLink"
                      data-toggle="dropdown"
                      aria-haspopup="true"
                      aria-expanded="false"
                    >
                      <i className="material-icons fa-2x">&#xE853;</i>
                    </a>
                  )}

                  <div
                    className="dropdown-menu dropdown-menu-right custom-dropdown"
                    aria-labelledby="navbarDropdownMenuLink"
                  >
                    <Navlinks user={this.props.auth} logOut={this.logOut} />
                  </div>
                </li>
              </ul>
              <ReactTooltip place="bottom" type="dark" effect="float" />
            </div>
          </div>
        </nav>
      </section>
    );
  }
}

const mapStateToProps = state => ({
  auth: state.user,
  user: state.user.userProfile
});

export default connect(mapStateToProps, { getProfile, signOut })(Navbar);
