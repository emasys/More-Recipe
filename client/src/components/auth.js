import jwt_decode from 'jwt-decode';

const login = localStorage.getItem('token');
let decoded = '';
/**
 *
 *
 * @class Auth
 */
class Auth {
  /**
   *
   *
   * @static
   * @memberof Auth
   */
  static logout() {
    localStorage.removeItem('token');
  }
  /**
 *
 *
 * @static
 * @returns
 * @memberof Auth
 */
  static loggedIn() {
    if (login) {
      decoded = jwt_decode(login);
      return true;
    }
    return false;
  }
  /**
 *
 *
 * @static
 * @returns
 * @memberof Auth
 */
  static userID() {
    return decoded.id;
  }
  /**
 *
 *
 * @static
 * @returns
 * @memberof Auth
 */
  static moniker() {
    return decoded.moniker;
  }
}

export default Auth;
