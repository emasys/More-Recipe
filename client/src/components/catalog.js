import { Link } from 'react-router-dom';
import React, { Component } from 'react';
// import $ from 'jquery';
// import { findDOMNode } from 'react-dom';
import Fade from 'react-reveal/Fade';

class Catalog extends Component {
  constructor(props) {
    super(props);

    this.state = {
      hover: ''
    };
    this.generateList = this.generateList.bind(this);
    this.hoverIn = this.hoverIn.bind(this);
    this.hoverOut = this.hoverOut.bind(this);
  }

  componentDidMount() {}

  hoverIn() {
    console.log('hovered');
  }

  hoverOut() {
    console.log('hovered out');
  }
  generateList({ catalog }) {
    if (catalog) {
      console.log(catalog);
      if (catalog.recipes.length < 1) {
        return (
          <div className="text-center error-message">
            <div className="catalog-wrapper">
              <img src="../img/logo.png" alt="logo" />
              <h4 className="p-3 m-2">
                We Know this is embarrassing, but we don't understand your query{' '}
              </h4>
              <p className="p-3 m-2">
                Perhaps you should cross-check your spellings
              </p>
            </div>
          </div>
        );
      }
      return catalog.recipes.map((item, index) => {
        return (
          <div
            key={index}
            className="col-lg-3 col-sm-10 mb-3  col-md-4 animate-catalog"
            data-animate="bounceIn"
            data-duration="1.0s"
            data-delay="0.1s"
            data-offset="100"
          >
            <div style={{ overflow: 'hidden' }}>
              <Fade bottom>
                <Link
                  to={`/recipe/${item.id}`}
                  onMouseEnter={this.hoverIn}
                  onMouseLeave={this.hoverOut}
                >
                  <div className={`card animate`}>
                    <img
                      className="card-img-top img-box"
                      src={`../../../img/uploads/${item.foodImg}`}
                      alt="Card image cap"
                    />

                    <div className="card-body p-0 text-center social-icons">
                      <a href="#">
                        <span className="tag bg-danger">{item.category}</span>
                      </a>
                      <h6 className="card-title custom-bg bg-dark p-2 m-0 text-truncate ">
                        {item.name}
                      </h6>
                      <div className="card-body p-0 text-center">
                        <p className="crop-text">{item.description}</p>
                      </div>
                      <span>
                        <i className="fa fa-heart-o" aria-hidden="true" />
                        {item.favorite}
                      </span>
                      <span>
                        <i className="fa fa-thumbs-o-up" aria-hidden="true" />
                        {item.upvote}
                      </span>
                      <span>
                        <i className="fa fa-thumbs-o-down" aria-hidden="true" />
                        {item.downvote}
                      </span>
                      <span>
                        <i className="fa fa-eye" aria-hidden="true" />
                        {item.views}
                      </span>
                      <span>
                        <i className="fa fa-comment-o" aria-hidden="true" />
                        {item.comments}
                      </span>
                    </div>
                  </div>
                </Link>
              </Fade>
            </div>
          </div>
        );
      });
    }
  }
  render() {
    return (
      <div className="row justify-content-center">
        {this.generateList(this.props)}
      </div>
    );
  }
}

export default Catalog;
