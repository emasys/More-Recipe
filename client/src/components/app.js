import React, { Component } from 'react';
// import Script from 'react-load-script';
// import Header from './header';
import Catalog from '../containers/catalog';
import { Link } from 'react-router-dom';
import { getRecipes, searchRecipes } from '../actions';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

//component
import CatalogList from '../components/catalog';
import Navbar from './navbar';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      search: '',
      All_recipes: '',
      page_limit: 12
    };

    // this.generateList = this.generateList.bind(this);
    this.searchBar = this.searchBar.bind(this);
    this.search = this.search.bind(this);
  }

  componentDidMount() {
    const query = '?sort=upvotes&order=desc';
    this.props.getRecipes(this.state.page_limit, query).then(() => {
      this.setState({
        All_recipes: this.props.recipes.allRecipes
      });
    });
  }

  search(e) {
    e.preventDefault();
    const data = { query: this.state.search };
    this.props.searchRecipes(data).then(() => {
      this.setState({
        All_recipes: this.props.recipes.search
      });
    });
  }
  searchBar(e) {
    this.setState({
      search: e.target.value
    });
    this.componentDidMount();
  }
  
  render() {
    const { search } = this.state;
    const headerImg = {
      background: `linear-gradient(90deg, rgba(241, 193, 89, 0.45), rgba(241, 193, 89, 0.5), rgba(241, 193, 89, 0.6), rgba(241, 193, 89, 0.7), rgba(241, 193, 89, 0.8), rgba(241, 193, 89, 0.9), rgba(241, 193, 89, 1)), 
      url('../img/Food.jpg') no-repeat center center`,
      backgroundSize: 'cover'
    };
    return (
      <div>
        <section className="container-fluid ">
          <div className="header" style={headerImg}>
            <Navbar />
            <div className="row header-items justify-content-center">
              <div className="col-lg-6 col-sm-10 text-center">
                <img src="../img/logo.png" alt="logo" />
                <p className=" text-shadowed bolden mt-2">
                  More-Recipes provides a platform for you to share the awesome
                  and exciting recipe ideas you have invented or learnt. feel
                  free to Browse through amazing recipes from around the world,
                  or &nbsp;<Link
                    to="/signup"
                    className="btn btn-dark hvr-icon-buzz-out"
                    role="button"
                  >
                    Sign Up&nbsp;
                  </Link>&nbsp;to post yours.
                </p>
              </div>
            </div>
          </div>
        </section>
        <section className="container catalog-wrapper" id="catalog">
          <div className="col-12">
            <div className="clearfix">
              <h4 className="float-left fresh-title">Top Recipes Today</h4>
              <h5 className="float-right">
                <Link
                  to="/catalog"
                  className="btn btn-dark hvr-icon-wobble-horizontal"
                >
                  see all recipes{` `}
                </Link>
              </h5>
            </div>
          </div>
          <CatalogList catalog={this.state.All_recipes} />
        </section>
      </div>
    );
  }
}

const mapStateToProps = state => {
  console.log(state.recipes);
  return { recipes: state.recipes };
};

const mapDispatchToProps = dispatch => ({
  ...bindActionCreators(
    {
      getRecipes,
      searchRecipes
    },
    dispatch
  )
});
export default connect(mapStateToProps, mapDispatchToProps)(App);
