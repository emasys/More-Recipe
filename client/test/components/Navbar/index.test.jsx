import React from 'react';
import { mount } from 'enzyme';
import Router from 'react-mock-router';


// components
import { Navbar } from '../../../src/components/Navbar';
import * as props from '../../__mocks__/navbarMocks';

let wrapperNotAuthenticated = null;
let wrapperAuthenticated = null;

describe('Test suite for Navbar component', () => {
  beforeEach(() => {
    wrapperNotAuthenticated = mount(<Router>
      <Navbar {...props.props} />
    </Router>);

    wrapperAuthenticated = mount(<Router>
      <Navbar {...props.propsAuth} />
    </Router>);
  });

  describe('View as an unauthenticated user', () => {
    it('Should render without errors', () => {
      expect(wrapperNotAuthenticated.length).toEqual(1);
      expect(wrapperNotAuthenticated.find('.nb').text()).toMatch('More Recipes');
      expect(wrapperNotAuthenticated.find('a#signUp').text()).toMatch('Sign up');
      expect(wrapperNotAuthenticated.find('a#signIn').text()).toMatch('Sign in');
    });

    it('should match snapshot', () => {
      expect(wrapperNotAuthenticated).toMatchSnapshot();
    });
  });

  describe('View as an authenticated user', () => {
    it('Should render without errors', () => {
      expect(wrapperAuthenticated.length).toEqual(1);
      expect(wrapperAuthenticated.find('.nb').text()).toMatch('More Recipes');
      expect(wrapperAuthenticated.find('.dropdown-header.text-center').text()).toMatch('Signed in as admin');
    });

    it('should match snapshot', () => {
      expect(wrapperAuthenticated).toMatchSnapshot();
    });
  });
});
