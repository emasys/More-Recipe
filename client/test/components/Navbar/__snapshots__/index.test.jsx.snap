// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for Navbar component View as an authenticated user should match snapshot 1`] = `
<MockRouter
  createHref={[Function]}
  location=""
  params={Object {}}
  path=""
  push={[Function]}
  replace={[Function]}
  url=""
>
  <Navbar
    auth={
      Object {
        "authInfo": Object {
          "isLoggedIn": true,
          "userId": 1,
          "username": "admin",
        },
        "isLoggedIn": true,
      }
    }
    className="bg-dark"
    getProfile={[Function]}
    signOut={[Function]}
    user={
      Object {
        "data": Object {
          "avatar": "http:avatar.com",
        },
      }
    }
  >
    <section
      className="container-fluid m-0 p-0"
    >
      <nav
        className="navbar navbar-expand-lg navbar-dark bg-dark"
        data-aos="flip-up"
      >
        <div
          className="container"
        >
          <Link
            className="navbar-brand bolder ml-3 text-orange"
            replace={false}
            to="/"
          >
            <a
              className="navbar-brand bolder ml-3 text-orange"
              onClick={[Function]}
            >
              <span
                className="nb"
              >
                More Recipes
              </span>
            </a>
          </Link>
          <button
            aria-controls="navbarSupportedContent"
            aria-expanded="false"
            aria-label="Toggle navigation"
            className="navbar-toggler"
            data-target="#navbarSupportedContent"
            data-toggle="collapse"
            type="button"
          >
            <i
              aria-hidden="true"
              className="fa fa-bars fa-2x text-orange"
            />
          </button>
          <div
            className="collapse navbar-collapse justify-content-end"
            id="navbarSupportedContent"
          >
            <ul
              className="navbar-nav"
            >
              <li>
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="nav-link "
                  data-tip="Search for recipes"
                  to="/catalog"
                >
                  <Route
                    path="/catalog"
                  >
                    <Link
                      aria-current={false}
                      className="nav-link "
                      data-tip="Search for recipes"
                      replace={false}
                      to="/catalog"
                    >
                      <a
                        aria-current={false}
                        className="nav-link "
                        data-tip="Search for recipes"
                        onClick={[Function]}
                      >
                        <i
                          className="material-icons fa-2x d-none d-lg-inline"
                        >
                          
                        </i>
                        <span
                          className="d-lg-none pl-3"
                        >
                          Search
                        </span>
                      </a>
                    </Link>
                  </Route>
                </NavLink>
              </li>
              <li
                className="nav-item"
              >
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="nav-link"
                  data-tip="Catalog"
                  to="/catalog"
                >
                  <Route
                    path="/catalog"
                  >
                    <Link
                      aria-current={false}
                      className="nav-link"
                      data-tip="Catalog"
                      replace={false}
                      to="/catalog"
                    >
                      <a
                        aria-current={false}
                        className="nav-link"
                        data-tip="Catalog"
                        onClick={[Function]}
                      >
                        <i
                          aria-hidden="true"
                          className="material-icons fa-2x  d-none d-lg-inline"
                        >
                          
                        </i>
                        <span
                          className="d-lg-none pl-3"
                        >
                          Catalog
                        </span>
                      </a>
                    </Link>
                  </Route>
                </NavLink>
              </li>
              <li
                className="nav-item"
              >
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="nav-link "
                  data-tip="Your favorites"
                  id="favorite"
                  to="/favorites"
                >
                  <Route
                    path="/favorites"
                  >
                    <Link
                      aria-current={false}
                      className="nav-link "
                      data-tip="Your favorites"
                      id="favorite"
                      replace={false}
                      to="/favorites"
                    >
                      <a
                        aria-current={false}
                        className="nav-link "
                        data-tip="Your favorites"
                        id="favorite"
                        onClick={[Function]}
                      >
                        <i
                          className="material-icons fa-2x red d-none d-lg-inline"
                        >
                          
                        </i>
                        <span
                          className="d-lg-none pl-3"
                        >
                          Favorites
                        </span>
                      </a>
                    </Link>
                  </Route>
                </NavLink>
              </li>
              <li
                className="nav-item dropdown"
              >
                <a
                  aria-expanded="false"
                  aria-haspopup="true"
                  className="nav-link dropdown-toggle"
                  data-toggle="dropdown"
                  href="#"
                  id="navbarDropdownMenuLink"
                >
                  <img
                    alt="avi"
                    className="fa-2x img-icon rounded-circle"
                    src="http:avatar.com"
                  />
                </a>
                <div
                  aria-labelledby="navbarDropdownMenuLink"
                  className="dropdown-menu dropdown-menu-right custom-dropdown"
                >
                  <NavLinks
                    logOut={[Function]}
                    user={
                      Object {
                        "authInfo": Object {
                          "isLoggedIn": true,
                          "userId": 1,
                          "username": "admin",
                        },
                        "isLoggedIn": true,
                      }
                    }
                  >
                    <div>
                      <h6
                        className="dropdown-header text-center"
                      >
                        Signed in as admin
                      </h6>
                      <div
                        className="dropdown-divider"
                      />
                      <Link
                        className="dropdown-item bold"
                        replace={false}
                        to="/profile/1"
                      >
                        <a
                          className="dropdown-item bold"
                          onClick={[Function]}
                        >
                          Your profile
                        </a>
                      </Link>
                      <Link
                        className="dropdown-item bold"
                        replace={false}
                        to="/favorites"
                      >
                        <a
                          className="dropdown-item bold"
                          onClick={[Function]}
                        >
                          Your favorites
                        </a>
                      </Link>
                      <Link
                        className="dropdown-item bold"
                        replace={false}
                        to="/new"
                      >
                        <a
                          className="dropdown-item bold"
                          onClick={[Function]}
                        >
                          Add new recipe
                        </a>
                      </Link>
                      <div
                        className="dropdown-divider"
                      />
                      <Link
                        className="dropdown-item bold"
                        id="logout"
                        onClick={[Function]}
                        replace={false}
                        to="/"
                      >
                        <a
                          className="dropdown-item bold"
                          id="logout"
                          onClick={[Function]}
                        >
                          Logout
                        </a>
                      </Link>
                    </div>
                  </NavLinks>
                </div>
              </li>
            </ul>
            <ReactTooltip
              effect="float"
              insecure={true}
              place="bottom"
              resizeHide={true}
              type="dark"
              wrapper="div"
            >
              <div
                className="__react_component_tooltip place-top type-dark "
                data-id="tooltip"
              />
            </ReactTooltip>
          </div>
        </div>
      </nav>
    </section>
  </Navbar>
</MockRouter>
`;

exports[`Test suite for Navbar component View as an unauthenticated user should match snapshot 1`] = `
<MockRouter
  createHref={[Function]}
  location=""
  params={Object {}}
  path=""
  push={[Function]}
  replace={[Function]}
  url=""
>
  <Navbar
    auth={
      Object {
        "authInfo": Object {
          "isLoggedIn": false,
          "userId": 1,
          "username": "admin",
        },
        "isLoggedIn": false,
      }
    }
    className="bg-dark"
    getProfile={[Function]}
    user={null}
  >
    <section
      className="container-fluid m-0 p-0"
    >
      <nav
        className="navbar navbar-expand-lg navbar-dark bg-dark"
        data-aos="flip-up"
      >
        <div
          className="container"
        >
          <Link
            className="navbar-brand bolder ml-3 text-orange"
            replace={false}
            to="/"
          >
            <a
              className="navbar-brand bolder ml-3 text-orange"
              onClick={[Function]}
            >
              <span
                className="nb"
              >
                More Recipes
              </span>
            </a>
          </Link>
          <button
            aria-controls="navbarSupportedContent"
            aria-expanded="false"
            aria-label="Toggle navigation"
            className="navbar-toggler"
            data-target="#navbarSupportedContent"
            data-toggle="collapse"
            type="button"
          >
            <i
              aria-hidden="true"
              className="fa fa-bars fa-2x text-orange"
            />
          </button>
          <div
            className="collapse navbar-collapse justify-content-end"
            id="navbarSupportedContent"
          >
            <ul
              className="navbar-nav"
            >
              <li>
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="nav-link "
                  data-tip="Search for recipes"
                  to="/catalog"
                >
                  <Route
                    path="/catalog"
                  >
                    <Link
                      aria-current={false}
                      className="nav-link "
                      data-tip="Search for recipes"
                      replace={false}
                      to="/catalog"
                    >
                      <a
                        aria-current={false}
                        className="nav-link "
                        data-tip="Search for recipes"
                        onClick={[Function]}
                      >
                        <i
                          className="material-icons fa-2x d-none d-lg-inline"
                        >
                          
                        </i>
                        <span
                          className="d-lg-none pl-3"
                        >
                          Search
                        </span>
                      </a>
                    </Link>
                  </Route>
                </NavLink>
              </li>
              <li
                className="nav-item"
              >
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="nav-link"
                  data-tip="Catalog"
                  to="/catalog"
                >
                  <Route
                    path="/catalog"
                  >
                    <Link
                      aria-current={false}
                      className="nav-link"
                      data-tip="Catalog"
                      replace={false}
                      to="/catalog"
                    >
                      <a
                        aria-current={false}
                        className="nav-link"
                        data-tip="Catalog"
                        onClick={[Function]}
                      >
                        <i
                          aria-hidden="true"
                          className="material-icons fa-2x  d-none d-lg-inline"
                        >
                          
                        </i>
                        <span
                          className="d-lg-none pl-3"
                        >
                          Catalog
                        </span>
                      </a>
                    </Link>
                  </Route>
                </NavLink>
              </li>
              <li
                className="nav-item"
              >
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="nav-link "
                  data-tip="Your favorites"
                  id="favorite"
                  to="/favorites"
                >
                  <Route
                    path="/favorites"
                  >
                    <Link
                      aria-current={false}
                      className="nav-link "
                      data-tip="Your favorites"
                      id="favorite"
                      replace={false}
                      to="/favorites"
                    >
                      <a
                        aria-current={false}
                        className="nav-link "
                        data-tip="Your favorites"
                        id="favorite"
                        onClick={[Function]}
                      >
                        <i
                          className="material-icons fa-2x red d-none d-lg-inline"
                        >
                          
                        </i>
                        <span
                          className="d-lg-none pl-3"
                        >
                          Favorites
                        </span>
                      </a>
                    </Link>
                  </Route>
                </NavLink>
              </li>
              <li
                className="nav-item dropdown"
              >
                <a
                  aria-expanded="false"
                  aria-haspopup="true"
                  className="nav-link "
                  data-toggle="dropdown"
                  href="#"
                  id="navbarDropdownMenuLink"
                >
                  <i
                    className="material-icons fa-2x"
                  >
                    
                  </i>
                </a>
                <div
                  aria-labelledby="navbarDropdownMenuLink"
                  className="dropdown-menu dropdown-menu-right custom-dropdown"
                >
                  <NavLinks
                    logOut={[Function]}
                    user={
                      Object {
                        "authInfo": Object {
                          "isLoggedIn": false,
                          "userId": 1,
                          "username": "admin",
                        },
                        "isLoggedIn": false,
                      }
                    }
                  >
                    <h6>
                      <Link
                        className="dropdown-item bold"
                        id="signIn"
                        replace={false}
                        to="/signin"
                      >
                        <a
                          className="dropdown-item bold"
                          id="signIn"
                          onClick={[Function]}
                        >
                          Sign in
                        </a>
                      </Link>
                      <Link
                        className="dropdown-item bold"
                        id="signUp"
                        replace={false}
                        to="/signup"
                      >
                        <a
                          className="dropdown-item bold"
                          id="signUp"
                          onClick={[Function]}
                        >
                          Sign up
                        </a>
                      </Link>
                    </h6>
                  </NavLinks>
                </div>
              </li>
            </ul>
            <ReactTooltip
              effect="float"
              insecure={true}
              place="bottom"
              resizeHide={true}
              type="dark"
              wrapper="div"
            >
              <div
                className="__react_component_tooltip place-top type-dark "
                data-id="tooltip"
              />
            </ReactTooltip>
          </div>
        </div>
      </nav>
    </section>
  </Navbar>
</MockRouter>
`;
